gadget/install/install.go

```diff
@@ -779,7 +779,7 @@ func FactoryReset(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string,
 			keyForRole[laidOut.Role] = encryptionKey
 		}
 		if options.Mount && onDisk.Label != "" && laidOut.HasFilesystem() {
-			if err := mountFilesystem(fsDevice, onDisk.Filesystem, filepath.Join(boot.InitramfsRunMntDir, onDisk.Label)); err != nil {
+			if err := mountFilesystem(fsDevice, laidOut.Filesystem, getMntPointForPart(laidOut.VolumeStructure)); err != nil {
```

Comment:
If there a unit test that we could tweak ? It obviously is spread-testing so less criticial but it would be nice to have this on the unit test level too (but can be a followup unbreaking master is more important it seems)

Reply:
It's a good point! Let me just add this to my queue and i'll take a look! But yes, let us get this merged first as we do have spread tests covering this for now.